imports/plugins/core/ui-navbar/client/containers/navbar.js

```diff
@@ -1,16 +1,45 @@
 import _ from "lodash";
 import { registerComponent, composeWithTracker } from "@reactioncommerce/reaction-components";
+import { Meteor } from "meteor/meteor";
 import { Reaction } from "/client/api";
 import NavBar from "../components/navbar";
 import { Media, Shops } from "/lib/collections";
 
 export function composer(props, onData) {
   const shop = Shops.findOne(Reaction.getShopId());
   const searchPackage = Reaction.Apps({ provides: "ui-search" });
+  const user = Meteor.user();
   let searchEnabled;
   let searchTemplate;
   let brandMedia;
 
+  let shops;
+
+  if (user && user.roles) {
+    // Get all shops for which user has roles
+    shops = Shops.find({
+      $and: [
+        { _id: { $in: Object.keys(user.roles) } },
+        { $or: [{ "workflow.status": "active" }, { _id: Reaction.getPrimaryShopId() }] }
+      ]
+    }).fetch();
+  }
+
+  /**
+   * handleShopSelectChange
+   * @method
+   * @summary Handle change in selected shop
+   * @param {script} event
+   * @param {String} shopId - selected shopId
+   * @since 1.5.8
+   * @return {void}
+  */
+  const handleShopSelectChange = (event, shopId) => {
+    // Test that shopId is a valid ID
```

Comment:
I am confused why we need to do this. Why would we get back in invalid Id? Also, there is a better test available to ensure that's it's actually an ID (there are more requirements than just being alphanumeric)